packages/dcos-integration-test/extra/resiliency/test_aws_cf.py

```diff
@@ -59,15 +78,18 @@ def get_running_agents(group_name):
     dcos_api_session.post('/mesos/machine/down', json=down_hosts).raise_for_status()
 
     # Wait for replacements
-    test_util.helpers.wait_for_len(partial(get_running_agents, 'SlaveServerGroup'), len(dcos_api_session.slaves), 600)
     test_util.helpers.wait_for_len(
-        partial(get_running_agents, 'PublicSlaveServerGroup'), len(dcos_api_session.public_slaves), 600)
+        partial(list, dcos_stack.public_agent_instances),
+        len(dcos_api_session.public_slaves),
+        600)
+    test_util.helpers.wait_for_len(
+        partial(list, dcos_stack.private_agent_instances),
+        len(dcos_api_session.slaves),
+        600)
 
     # Reset the dcos_api_session to have the replacement agents
-    dcos_api_session.slaves = sorted([agent.private_ip_address for agent in
-                                      get_running_agents('SlaveServerGroup')])
-    dcos_api_session.public_slaves = sorted([agent.private_ip_address for agent in
-                                             get_running_agents('PublicSlaveServerGroup')])
+    dcos_api_session.slaves = get_private_ips(dcos_stack.private_agent_instances)
```

Comment:
Sure if you do it as part of another clean up that would work, I will approve this.